src/components/ReviewPost/ReviewPost.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from "react";
+import InputBox from "../InputBox/InputBox";
+import PreviewBox from "../PreviewBox/PreviewBox";
+import Button from "../Button/Button";
+
+const ReviewPost = ({ previewText, setPreviewText, setActivePage }) => {
+  const [isEditing, setIsEditing] = useState(false);
+
+  const handleBackClick = () => {
+    setActivePage("edit");
+  };
+
+  const handleToggleEditing = () => {
+    setIsEditing(!isEditing);
+  };
+
+  const previewContent = () => {
+    return isEditing ? (
+      <InputBox
+        value={previewText}
+        onChange={(e) => setPreviewText(e.target.value)}
+        className="promptpage__preview-box"
+      />
+    ) : (
+      <PreviewBox
+        previewText={previewText}
+        className="promptpage__preview-box"
+        onClickEdit={handleToggleEditing}
+      />
+    );
+  };
+
+  // placeholder to handle LinkedIn posting
+  const handlePostButtonClick = () => {};
```

Comment:
Maybe we'll also need to update the activePage state here, though I know this is just a placeholder for now.

Reply:
Ah yes, Mohan's functions might take care of that for us too - I need to take a look